### DISCUSSION TITLE
Control where alembic writes the alembic_version table

### DISCUSSION THREAD

**Discussion by Johnb6262** [sqlalchemy/sqlalchemy#11213] [Usage Questions] [answered]
I'm responsible for setting up alembic to run migrations on a new AI project.  I just ran locally against Postgres and noticed that the alembic_version table appears to go into the "public" database, not the schema where all my tables were created.  My architect wants to know if I can change that.

Is there a way to tell alembic to store that version table in the same schema it is operating on?

Thanks.

**Comment by zzzeek** [comment]
see https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.configure.params.version_table_schema

**Comment by Johnb6262** [comment]
That documentation is voluminous and does not give examples of use.  What is the simplest way to accomplish this?  Is it something that can be added to the ini file or the env file or does code have to be written somewhere?

**Reply by zzzeek** [reply]
that's reference documentation.   for a basic tutorial which covers where the environment options are placed, see https://alembic.sqlalchemy.org/en/latest/tutorial.html#the-migration-environment.   this option goes into the env.py file inside of context.configure().

**Comment by Johnb6262** [comment]
Thanks - however there is nothing on that page when I search "version_table_schema"...

Also, per some other advice I found I tried putting "version_table_schema" into the ini file (inside the [alembic] section) and it fails every time.

BUT if I add version_table_schema in the env.py - in the run_migrations_online method like below, it works. 

    with connectable.connect() as connection:
        context.configure(
            connection=connection,
            target_metadata=target_metadata,
            version_table_schema="chat_svg"
        )

So I have a solution, but I'd really like to understand why that isn't being picked up from the alembic.ini file...  Any ideas?

And yes, my ini file is at the root of my project which is where I issue the alembic -c alembic.ini upgrade head command (I tried specifying the file too, just to be sure)

**Reply by zzzeek** [reply, marked as answer]
> Thanks - however there is nothing on that page when I search "version_table_schema"...

once you understand how to configure your env.py, you'd consult the reference documentation for all the available options.


> Also, per some other advice I found I tried putting "version_table_schema" into the ini file (inside the [alembic] section) and it fails every time.

that would be incorrect advice (I'm assuming ChatGPT, since it looks like random elements strung together).   if chatgpt is giving you crappy information (disregard if this is not chatgpt), that's sort of what it's made for, we'd hardly want people coming *here* where the humans are and then complaining about AI tools.   

> BUT if I add version_table_schema in the env.py - in the run_migrations_online method like below, it works.

that's what I said you should do!   again assuming this is chatgpt advice (which it really sounds like, apologies if it's not)  human: 1 , chatgpt: 0

> So I have a solution, but I'd really like to understand why that isn't being picked up from the alembic.ini file... Any ideas?

This is not an alembic.ini option. if someone told you that, they're incorrect!   

if you'd **like** it to be an alembic.ini option, you can do that also, in your env.py you are given the configuration object in the ``config`` variable, you can consume custom options from it and apply them in your env.py.

**Comment by Johnb6262** [comment]
OK - so after looking more closely, this just is not picked up in the ini file.  Got it. Thanks.
